src/Foundation/NSUrl.cs

```diff
@@ -30,6 +30,32 @@ public NSUrl (string path, string relativeToUrl)
 		{
 		}
 
+		public NSUrl (string urlString)
+		{
+			Handle = _Init (urlString);
+			if (Handle == IntPtr.Zero)
+				throw new ArgumentException ("Malformed url");
+		}
+
+		public NSUrl (string urlString, NSUrl relativeToUrl)
+		{
+			Handle = _Init (urlString, relativeToUrl);
+			if (Handle == IntPtr.Zero)
+				throw new ArgumentException ("Malformed url");
+		}
+
+		// Call these instead of obj-c static versions since these throw
+		// if nil returned due to malformed input
```

Comment:
really ? I seem to recall it returns `null` if invalid - and this change could break application that depended on this (as a non-throw way to get an `NSUrl`)